pkg/datapath/loader/loader.go

```diff
@@ -135,15 +156,34 @@ func nullifyStringSubstitutions(strings map[string]string) map[string]string {
 	return nullStrings
 }
 
+func (l *loader) bpfMasqAddrs(ifName string) (ipv4, ipv6 netip.Addr) {
+	if l.cfg.DeriveMasqIPAddrFromDevice != "" {
```

Comment:
We're doing a one-off query here for the addresses. These of course can change later which would require reconciliation (which is missing from this PR). This will be done in a later PR that @dylandreimerink is working on. (We'll likely pull the query to the "top-half" of the loader and have a reconciliation loop there). Here as well I'm open to doing that refactoring already by instead passing the BPF masquerade addresses in some state struct down to `patchHostNetdevDatapath`. For now I opted not to do that yet as the design isn't entirely baked yet for it.